src/main/index.ts

```diff
@@ -1,13 +1,13 @@
 import * as Electron from "electron";
-import mitt from "mitt";
-import { platform } from "os";
-import * as Core from "./Core";
-import * as Extensions from "./Extensions";
 
-(async () => {
+const main = async () => {
     await Electron.app.whenReady();
 
-    Core.SingleInstanceLockModule.bootstrap(Electron.app);
+    const mitt = (await import("mitt")).default;
+    const platform = (await import("os")).platform;
+    const Core = await import("./Core");
+    const Extensions = await import("./Extensions");
+
```

Comment:
@oliverschwendener Is the switch to async imports here alright? Importing synchronously means ueli hangs for about 800ms (on my machine) before sending the second instance event to the first instance.

Reply:
I did some performance testing using node's `perf_hooks` module. To me it does not look like the import takes so long.

- The `import` statements takes around 0.07ms
- Waiting for `app.whenReady()` takes around 40ms
- Waiting for `requestSingleInstanceLock` takes around 0.75ms
- Bootstrapping the electron modules takes around 0.75ms
- Bootstrapping the core modules takes around 40ms

Reply:
Can you do some measuring and see if it behaves similarly on your machine?

Reply:
I tried two different methods (apologies in advance for the messy code)
- [ES6 Import](https://github.com/ke1v/ueli/blob/99661c30cddefb5fc0fc0241507641bb21fd60a5/src/main/index.ts)
    - **SingleInstanceLock**
     Starts ~750ms (Import time)
     Duration ~110ms (Electron start time)
- [Dynamic Import](https://github.com/ke1v/ueli/blob/f6df85858a86f39a48115db4f4ffc50a276281fb/src/main/index.ts)
    - **Import**
     Starts ~180ms
     Duration ~700ms
    - **SingleInstanceLock**
     Starts ~900ms
     Duration 0.3ms

Both tests seems to suggest that `import` takes ~700ms. `app.whenReady()` and `requestSingleInstanceLock` takes about 100ms and 0.3ms which lines up with your testing. How did you time the import statements? 

Tested on a Ryzen 2600